chips/sam4l/src/adc.rs

```diff
@@ -393,7 +427,12 @@ impl hil::adc::Adc for Adc {
     fn sample_continuous(&self, channel: &Self::Channel, frequency: u32) -> ReturnCode {
         let regs: &mut AdcRegisters = unsafe { mem::transmute(self.registers) };
 
-        if !self.enabled.get() {
+        let res = self.config_and_enable(frequency);
```

Comment:
Why are you only implementing this logic for `sample_continuous` and not for `sample_highspeed` as well?